client/hosting/hosting-features/components/hosting-features.tsx

```diff
@@ -58,34 +58,35 @@ const HostingFeatures = () => {
 	);
 	const hasEnTranslation = useHasEnTranslation();
 
-	const transfer = useSelector( ( state ) => getAtomicTransfer( state, siteId ) );
-	const isTransferInProgress = [
-		transferStates.PENDING,
-		transferStates.ACTIVE,
-		transferStates.PROVISIONED,
-	].includes( transfer.status );
+	const { data: siteTransferData, refetch: refetchSiteTransferData } = useSiteTransferStatusQuery(
+		siteId || undefined
+	);
+	// We have some bug at the backend, when the transfer is complited, but isSiteAtomic is still false, so we need such extra condition
+	const isTransferInProgress =
+		siteTransferData?.isTransferring || siteTransferData?.status === transferStates.COMPLETED;
 
 	useEffect( () => {
 		if ( ! siteId ) {
 			return;
 		}
 
-		if ( transfer.status === transferStates.COMPLETED ) {
-			window.location.href = `/overview/${ siteSlug }`;
-			return;
-		}
-
-		dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
-		// A user can keep one tab open and start transfer in the second tab
-		// se we always run interval to check the status, to avoid case with rendering "Activate" button on old tabs of a browser
 		const interval = setInterval( () => {
-			dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
-		}, 10000 );
+			if ( siteTransferData?.status !== transferStates.COMPLETED ) {
+				refetchSiteTransferData();
+			} else {
+				clearInterval( interval );
+				dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
+			}
+		}, 3000 );
 
-		return () => {
-			clearInterval( interval );
-		};
-	}, [ siteSlug, siteId, transfer.status, dispatch ] );
+		return () => clearInterval( interval );
+	}, [ siteId, siteTransferData?.status, refetchSiteTransferData, dispatch ] );
+
+	useEffect( () => {
+		if ( isSiteAtomic && ! isPlanExpired ) {
+			page.replace( redirectUrl.current );
```

Comment:
As I mentioned in https://github.com/Automattic/wp-calypso/pull/93488#issuecomment-2288821664, after the transfer is completed, I am usually automatically redirected to `/wp-login.php` for the site. I only managed to land on `/overview` _once_, and I don't know what differentiated that test from my other ones.

Did you also experience this, @nightnei?

Reply:
Just not to leave open thread, I would duplicate my answer from Slack - I experience it too, also randomly, and with both A4A and Simple->Atomic. Also, I experienced it frequently too, so I just accepted that some times necessary to authorize after getting Atomic website. And it's one of the points which made development here tough.